CONTRIBUTING.md

```diff
@@ -17,19 +17,70 @@ Before submitting a new issue, do the following:
 - Verify you have Xcode tools installed by running `xcode-select --install`.
 - Make sure to read through the [README](https://github.com/fastlane/fastlane) of the project.
 
-
 When submitting a new issue, please provide the following information:
 
 - The full stack trace and output when running `fastlane`.
 - The command and parameters you used to launch it.
-- Your `Fastfile` and all other configuration files you are using. 
+- Your `Fastfile` and all other configuration files you are using.
 
 *By providing this information it's much faster and easier to help you*
 
-## Pull Requests
+## Helping to Resolve Existing Issues
+
+If you're motivated to help out at a level beyond reporting issues, we really appreciate it! :+1: A good place to start is by reviewing the list of [open issues](https://github.com/fastlane/fastlane/issues) or [open PRs](https://github.com/fastlane/fastlane/pulls) for things that need attention. When you find one that you'd like to help with, the following things are super valuable:
+
+### Verifying Bug Reports
+
+Are you able to reproduce this problem on your computer? If so, chime in! If the original issue report is a bit vague, but you have some additional details, please contribute them! If you find a bug report without an example script or configuration, contribute one that can be used to reproduce it.
+
+If you're comfortable diving into the code a bit, see if you can produce a failing test that illustrates the bad behavior in the problematic area of the code! Tests live in the `spec/` directory, and are generally named relative to the source code files whose functionality they test. Once you are done, `git diff > my_patch.diff` can be used to produce a patch file containing the details of your changes, which can be added to the issue.
+
+### Testing Proposed Fixes
+
+Another way to help out is to verify submitted pull requests. To do that, you'll need to be able to get the author's proposed changes onto your machine. Start by giving yourself a new branch to work in:
+
+```
+git checkout -b testing_branch
+```
+
+Next, you'll need to tell git where to find this contributor's fastlane fork and branch. Let's say that the contributor's username is **FrauseKx** and their topic branch is called `new_fastlane_action` located at **https://github.com/FrauseKx/fastlane**. You can use the following commands to pull their work:
```

Comment:
Then let's replace it with `[user]` to make it clear :+1: People might still copy the user